mirrulations-validation/src/mirrval/job_validator.py

```diff
@@ -1,54 +1,69 @@
 import os
 import sys
 import time
-import dotenv
+from collections import Counter
+import json
+from dotenv import load_dotenv
 import redis
 from mirrgen.search_iterator import SearchIterator
 from mirrcore.redis_check import is_redis_available
 from mirrcore.regulations_api import RegulationsAPI
 from mirrcore.data_storage import DataStorage
-from mirrcore.job_queue import JobQueue
 
 
 class Validator:
 
-    def __init__(self, api, datastorage, job_queue):
+    def __init__(self, api, datastorage):
         self.api = api
         self.datastorage = datastorage
-        self.job_queue = job_queue
+        self.unfound_jobs = {}
 
     def download(self, endpoint):
         beginning_timestamp = '1972-01-01 00:00:00'
         collection_size = self.datastorage.get_collection_size(endpoint)
-        counter = 0
+        counter = Counter()
         for result in SearchIterator(self.api, endpoint, beginning_timestamp):
             if result == {}:
                 continue
             for res in result['data']:
                 if not self.datastorage.exists(res):
-                    print(res['id'])
-                    self.job_queue.add_job(res['links']['self'], res['type'])
-                counter += 1
-            percentage = (counter / collection_size) * 100
-            print(f'{percentage:.2f}%')
+                    print(f"{res['id']} not in database, writing to file")
+                    write_unfound_jobs(res, self.unfound_jobs)
+                    time.sleep(3.6)
+                    counter['Not_in_db'] += 1
+                else:
+                    print(f"{res['id']} exists in database")
```

Comment:
This may clog up the logs. We may only want to know what files do not exist.

Reply:
Good point, resolving now.